.github/workflows/check-package.yml

```diff
@@ -11,20 +11,29 @@ on:
         description: 'PyPI name to install the package with'
         required: true
         type: string
+      test-matrix:
+        description: 'what test configs to run in json format'
+        required: false
+        type: string
+        default: |
```

Comment:
What's the benefit of specifying that compared to a usual matrix?

Reply:
good call, I may stay also with having the workflow here simple without branching...

Reply:
> What's the benefit of specifying that compared to a usual matrix?

It enables each project to specify the matrix on their own.

In most cases, we can just ignore `test-matrix` and use the default value, but for example in Flash, Windows is excluded from the matrix, and so there needs to be a way to customise the matrix to be able to reuse this workflow:
https://github.com/Lightning-AI/lightning-flash/blob/04f1b8f59847b8d49b96b447e42d5e14aff966b8/.github/workflows/ci-install-pkg.yml#L35-L36

Otherwise, I don't think there's a way to reuse this workflow in Flash, and we then need to copy and paste the content of `check-package.yml` and exclude Windows there, which stands against the purpose of this repository.

Reply:
@justusschock @Borda Could you leave your thoughts/concerns if you have when you get a chance?